spec/numberformat.html

```diff
@@ -104,54 +104,98 @@
+            1. Add new part record { [[type]]: *"currency"*, [[value]]: _cd_ } as a new element of the list _result_.
           1. Else,
-            1. Let _n_ be ToRawFixed(_x_, _numberFormat_.[[minimumIntegerDigits]], _numberFormat_.[[minimumFractionDigits]], _numberFormat_.[[maximumFractionDigits]]).
-          1. If the value of the _numberFormat_.[[numberingSystem]] matches one of the values in the "Numbering System" column of <emu-xref href="#table-2">Table 2</emu-xref> below, then
-            1. Let _digits_ be an array whose 10 String valued elements are the UTF-16 string representations of the 10 _digits_ specified in the "Digits" column of the matching row in <emu-xref href="#table-2">Table 2</emu-xref>.
-            1. Replace each _digit_ in _n_ with the value of _digits_[_digit_].
-          1. Else use an implementation dependent algorithm to map _n_ to the appropriate representation of _n_ in the given numbering system.
-          1. If _n_ contains the character *"."*, replace it with an ILND String representing the decimal separator.
-          1. If the value of the _numberFormat_.[[useGrouping]] is *true*, insert an ILND String representing a grouping separator into an ILND set of locations within the integer part of _n_.
-        1. If _negative_ is *true*, then
-          1. Let _result_ be the value of _numberFormat_.[[negativePattern]].
-        1. Else,
-          1. Let _result_ be the value of _numberFormat_.[[positivePattern]].
-        1. Replace the substring *"{number}"* within _result_ with _n_.
-        1. If the value of the _numberFormat_.[[style]] is *"currency"*, then
-          1. Let _currency_ be the value of _numberFormat_.[[currency]].
-          1. If _numberFormat_.[[currencyDisplay]] is *"code"*, then
-            1. Let _cd_ be _currency_.
-          1. Else if _numberFormat_.[[currencyDisplay]] is *"symbol"*, then
-            1. Let _cd_ be an ILD string representing _currency_ in short form. If the implementation does not have such a representation of _currency_, use _currency_ itself.
-          1. Else if _numberFormat_.[[currencyDisplay]] is *"name"*, then
-            1. Let _cd_ be an ILD string representing _currency_ in long form. If the implementation does not have such a representation of _currency_, then use _currency_ itself.
-          1. Replace the substring *"{currency}"* within _result_ with _cd_.
+            1. Add new part record { [[type]]: *"literal"*, [[value]]: _p_ } as a new element of the list _result_.
+          1. Set _index_ to _endIndex_ + 1.
+          1. Set _beginIndex_ to Call(*%StringProto_indexOf%*, _pattern_, "}", _index_)
+        1. If _endIndex_ is less than _length_, then:
+          1. Let _literal_ be the substring of _pattern_ from position _endIndex_, exclusive, to position _length_, exclusive.
+          1. Add new part record { [[type]]: *"literal"*, [[value]]: _literal_ } as a new element of the list _result_.
```

Comment:
This should use `_index_` instead of `_endIndex_` and the substring should be from `position _index_, inclusive, to position _length_, exclusive.`  Otherwise, an empty `literal` is produced at the end.